gui/gui/src/ts/components/search/NodeSearchBox.tsx

```diff
@@ -1,41 +1,68 @@
 import * as React from "react";
+import * as qs from "qs";
 
 import {Paper, InputAdornment, InputBase} from "@material-ui/core";
 
 import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
 import {faSearch} from "@fortawesome/free-solid-svg-icons";
+import {useHistory} from "react-router-dom";
+import {Hrefs} from "Hrefs";
 
 export const NodeSearchBox: React.FunctionComponent<{
-  onSubmit: (value: string) => void;
-}> = ({onSubmit}) => {
-  const [search, setSearch] = React.useState<{text: string}>({text: ""});
+  placeholder?: string;
+  showIcon?: boolean;
+  onSubmit?: (value: string) => void;
+  style?: React.CSSProperties;
+  value?: string;
+  onChange?: (value: string) => void;
+}> = ({onSubmit, showIcon = false, placeholder, style, value, onChange}) => {
+  const [search, setSearch] = React.useState<{text: string}>({
+    text: value || "",
+  });
+
+  const history = useHistory();
+
+  if (!onSubmit) {
+    onSubmit = (text: string) => {
```

Comment:
Reassigning props is poor form. What I usually do is rename the variable in the (...) and then use the name in the body. Like:

```
}> = ({onSubmit: onSubmitUserDefined, ...}) => {
  const onSubmit = onSubmitUserDefined ?? (text:string) => ...
}
```

Reply:
In general variables should be `const` (single assignment) unless there is a strong reason to do otherwise.
This makes reading the code easier and helps the compiler. (At a very low level, the compiler translates multiple assignments to single assignments, anyway: https://en.wikipedia.org/wiki/Static_single_assignment_form)

Reply:
Gotcha will keep in mind, changed